modules/SofaMiscFem/src/SofaMiscFem/QuadBendingFEMForceField.inl

```diff
@@ -0,0 +1,1261 @@
+  Ks[18][17]=length*height*(Js_t[18][30]*Cs[6][6]*Js[30][17])*ws*ws;
+ 
+  Ks[18][3]=length*height*(Js_t[18][27]*Cs[3][3]*Js[3][3]+Js_t[18][29]*Cs[5][5]*Js[5][3]+Js_t[18][30]*Cs[6][6]*Js[6][3])*ws*ws;
+  Ks[18][8]=length*height*(Js_t[18][27]*Cs[3][3]*Js[11][8]+Js_t[18][29]*Cs[5][5]*Js[13][8]+Js_t[18][30]*Cs[6][6]*Js[14][8])*ws*ws;
+  Ks[18][13]=length*height*(Js_t[18][27]*Cs[3][3]*Js[19][13]+Js_t[18][29]*Cs[5][5]*Js[21][13]+Js_t[18][30]*Cs[6][6]*Js[22][13])*ws*ws;
+  Ks[18][18]=length*height*(Js_t[18][27]*Cs[3][3]*Js[27][18]+Js_t[18][29]*Cs[5][5]*Js[29][18]+Js_t[18][30]*Cs[6][6]*Js[30][18])*ws*ws;
+
+  Ks[18][4]=length*height*(Js_t[18][27]*Cs[3][4]*Js[4][4]+Js_t[18][29]*Cs[5][5]*Js[5][4])*ws*ws;
+  Ks[18][9]=length*height*(Js_t[18][27]*Cs[3][4]*Js[12][9]+Js_t[18][29]*Cs[5][5]*Js[13][9])*ws*ws;
+  Ks[18][14]=length*height*(Js_t[18][27]*Cs[3][4]*Js[20][14]+Js_t[18][29]*Cs[5][5]*Js[21][14])*ws*ws;
+  Ks[18][19]=length*height*(Js_t[18][27]*Cs[3][4]*Js[28][19]+Js_t[18][29]*Cs[5][5]*Js[29][19])*ws*ws;
+
+  Ks[19][2]=length*height*(Js_t[19][31]*Cs[7][7]*Js[7][2])*ws*ws;
+  Ks[19][7]=length*height*(Js_t[19][31]*Cs[7][7]*Js[15][7])*ws*ws;
+  Ks[19][12]=length*height*(Js_t[19][31]*Cs[7][7]*Js[23][12])*ws*ws;
+  Ks[19][17]=length*height*(Js_t[19][31]*Cs[7][7]*Js[31][17])*ws*ws;
+
+  Ks[19][3]=length*height*(Js_t[19][28]*Cs[4][3]*Js[3][3]+Js_t[19][29]*Cs[5][5]*Js[5][3])*ws*ws;
+  Ks[19][8]=length*height*(Js_t[19][28]*Cs[4][3]*Js[11][8]+Js_t[19][29]*Cs[5][5]*Js[13][8])*ws*ws;
+  Ks[19][13]=length*height*(Js_t[19][28]*Cs[4][3]*Js[19][13]+Js_t[19][29]*Cs[5][5]*Js[21][13])*ws*ws;
+  Ks[19][18]=length*height*(Js_t[19][28]*Cs[4][3]*Js[27][18]+Js_t[19][29]*Cs[5][5]*Js[29][18])*ws*ws;
+
+  Ks[19][4]=length*height*(Js_t[19][28]*Cs[4][4]*Js[4][4]+Js_t[19][29]*Cs[5][5]*Js[5][4]+Js_t[19][31]*Cs[7][7]*Js[7][4])*ws*ws;
+  Ks[19][9]=length*height*(Js_t[19][28]*Cs[4][4]*Js[12][9]+Js_t[19][29]*Cs[5][5]*Js[13][9]+Js_t[19][31]*Cs[7][7]*Js[15][9])*ws*ws;
+  Ks[19][14]=length*height*(Js_t[19][28]*Cs[4][4]*Js[20][14]+Js_t[19][29]*Cs[5][5]*Js[21][14]+Js_t[19][31]*Cs[7][7]*Js[23][14])*ws*ws;
+  Ks[19][19]=length*height*(Js_t[19][28]*Cs[4][4]*Js[28][19]+Js_t[19][29]*Cs[5][5]*Js[29][19]+Js_t[19][31]*Cs[7][7]*Js[31][19])*ws*ws;
+
+
+
+  // Stiffness matrix of a element: K = Kb + Ks
+  K = Kb + Ks;
+  // save stiffness
+  quadInf[elementIndex].Bendingstiffness=Kb;
+  quadInf[elementIndex].Shearstiffness=Ks;
+  quadInfo.endEdit();
+
+ /*int nbQuads = m_topology->getNbQuads();
+    //count number line of file 
+    int numLines = 0;
+    std::ifstream in("/home/nhnhanbk/Desktop/Sofa/sofa/nhnhan/IoTouch/K.txt");
+    std::string unused;
+    while (std::getline(in, unused))
+        ++numLines;
+    in.close();
+
+    if (numLines == nbQuads) {std::ofstream myfile("/home/nhnhanbk/Desktop/Sofa/sofa/nhnhan/IoTouch/K.txt", std::ios_base::app);
+				myfile.close();
+				}
+	else
+    {
+        std::ofstream myfile("/home/nhnhanbk/Desktop/Sofa/sofa/nhnhan/IoTouch/K.txt", std::ios_base::app);
```

Comment:
If you think output file is an interesting feature you could create a Datafilename variable and set that path to the ofstream. Otherwise could you remove this commented block.

Reply:
I tried to add a variable (std::string **directory**) for all users to add their own path (see the lasted update), however, It did not work because I can not either convert **directory** variable (string) to char type as requested of ofstream. Do you have any idea how to do it? It is quite important information for the users of our plugin.